tests/js/client/aql/aql-mdi.js

```diff
@@ -309,19 +383,139 @@ function optimizerRuleMdi2dIndexTestSuite() {
     testCreateAsZkd: function () {
       let col = db._create(colName + "5");
       const idx = col.ensureIndex({
-        type: 'zkd',
+        type: "zkd",
         name: "myIndex",
-        fields: ['x', 'y'],
-        fieldValueTypes: 'double',
+        fields: ["x", "y"],
+        fieldValueTypes: "double",
       });
-      assertEqual('zkd', idx.type);
+      assertEqual("zkd", idx.type);
 
       const idx2 = col.index("myIndex");
       assertEqual(idx2.id, idx.id);
-      assertEqual(idx2.type, 'zkd');
+      assertEqual(idx2.type, "zkd");
 
       col.drop();
-    }
+    },
+
+    testInRangeNonStrict: function () {
+      const queries = [
+        aql`FOR d IN ${col} FILTER 0 < d.x && d.x < 1 RETURN d.x`,
+        aql`FOR d IN ${col} FILTER IN_RANGE(d.x, 0, 1, false, false) RETURN d.x`,
+      ];
+
+      for (let i = 0; i < queries.length; ++i) {
+        const query = queries[i];
+        const explainRes = db
+          ._createStatement({ query: query.query, bindVars: query.bindVars })
+          .explain();
+        const appliedRules = explainRes.plan.rules;
+        const nodeTypes = explainRes.plan.nodes
+          .map((n) => n.type)
+          .filter((n) => !["GatherNode", "RemoteNode"].includes(n));
+        assertEqual(["SingletonNode", "IndexNode", "ReturnNode"], nodeTypes);
+        assertTrue(appliedRules.includes(useIndexes));
+        assertFalse(appliedRules.includes(removeFilterCoveredByIndex));
+        const executeRes = db._query(query.query, query.bindVars);
```

Comment:
I would like to check also, that there is no post filter in the index node. I don't see a check that ensures, that the post filter is indeed removed, except from checking if the rule `removeFilterCoveredByIndex` was executed. But that rule might have done anything but what we want.